AWSSDKSwiftCLI/Sources/SPR/UploadArchive.swift

```diff
@@ -0,0 +1,60 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import AWSS3
+import Smithy
+import SmithyStreams
+import AWSCLIUtils
+
+extension SPRPublisher {
+
+    mutating func uploadArchive() async throws {
```

Comment:
This function:
- Creates a new .zip archive of a Swift package,
- Calculates the SHA-256 checksum for the archive,
- Gets the archive from S3 to verify that it does not already exist, and
- Puts the archive to S3.